### PR TITLE
OTLP Span attributes fixes

### PR DESCRIPTION
**Pull request by pcwiese**
## Changes
- Coerce Activity tags into the correct OTLP attribute types
- Ensure that the attribute type is properly specified

At the very least, we need to set the attribute type, but I also think it is
important to use the attribute types accurately.

### Checklist
- [X] I ran Unit Tests locally.
- [X] I verified on the receiving end using the latest collector

The PR will automatically request reviews from [code owners](https://github.com/open-telemetry/opentelemetry-dotnet/blob/master/CODEOWNERS#L15) and trigger CI build and tests.

### CODE REVIEW

**Comment by pcwiese** [comment]
@pjanotti Can you sanity check this?

File: src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/ActivityExtensions.cs [hunk 1]
```diff
@@ -202,25 +197,79 @@ private static OtlpTrace.Span.Types.Event ToOtlpEvent(ActivityEvent activityEven
             return otlpEvent;
         }
 
+        private static OtlpCommon.AttributeKeyValue ToOtlpAttribute(KeyValuePair<string, string> kvp)
+        {
+            // TODO: enforce no duplicate keys?
```
Comment (cijothomas) [inline src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/ActivityExtensions.cs:202]: .NET team if making Activity.Attributes to match OT spec - so the order will be maintained, duplicated will be prevented etc. 

https://github.com/dotnet/designs/pull/134/files#diff-1f85dcaf1be8ff277864f69c473c6b33R34

Comment (pcwiese) [reply src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/ActivityExtensions.cs:202]: Great. This logic will need to be touched again once those activity changes are made by them. Looks like we will be able to utilize the Activity.Attributes when it becomes available

Comment (cijothomas) [reply src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/ActivityExtensions.cs:202]: Yes.

File: src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/ActivityExtensions.cs [hunk 2]
```diff
@@ -202,25 +197,79 @@ private static OtlpTrace.Span.Types.Event ToOtlpEvent(ActivityEvent activityEven
...
         }
 
+        private static OtlpCommon.AttributeKeyValue ToOtlpAttribute(KeyValuePair<string, string> kvp)
+        {
+            // TODO: enforce no duplicate keys?
+            // TODO: reverse?
+            // To maintain full fidelity to downstream receivers convert to the proper attribute types
+            var attrib = new OtlpCommon.AttributeKeyValue { Key = kvp.Key };
+
+            if (kvp.Value == null)
```
Comment (pjanotti) [inline src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/ActivityExtensions.cs:207]: OTel spec for attributes indicates that the key should be removed if the value is null:
> Attribute values of null are considered to be not set and get discarded as if that SetAttribute call had never been made. As an exception to this, if overwriting of values is supported, this results in clearing the previous value and dropping the attribute key from the set of attributes.

from https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/api.md#set-attributes

**Comment by pcwiese** [comment]
No changes in the last iteration, just some squashing. There is flaky test I think in an unrelated project. Is there a way for contributors to trigger the test pipeline manually?

**Comment by cijothomas** [comment]
> No changes in the last iteration, just some squashing. There is flaky test I think in an unrelated project. Is there a way for contributors to trigger the test pipeline manually?

One easy trick is to simply close/open the PR, which will re-trigger all checks/CI.

**Pull request merged**